lib/improver/tests/lapse_rate/test_apply_lapse_rate.py

```diff
@@ -0,0 +1,168 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the apply_lapse_rate method."""
+
+import numpy as np
+import unittest
+
+import iris
+from iris.tests import IrisTest
+
+from improver.lapse_rate import apply_lapse_rate
+from improver.constants import DALR
+from improver.tests.set_up_test_cubes import (
+    set_up_variable_cube, add_coordinate)
+
+
+class test_apply_lapse_rate(IrisTest):
+    """Test the apply_lapse_rate method"""
+
+    def setUp(self):
+        """Set up some input cubes"""
+        source_orog = np.array([[400., 400., 402., 402.],
+                                [400., 400., 402., 402.],
+                                [403., 403., 405., 405.],
+                                [403., 403., 405., 405.]], dtype=np.float32)
+        self.source_orog = set_up_variable_cube(
+            source_orog, name='orography', units='m', spatial_grid='equalarea')
+
+        dest_orog = np.array([[400., 401., 401., 402.],
+                              [402., 402., 402., 403.],
+                              [403., 404., 405., 404.],
+                              [404., 405., 406., 405.]], dtype=np.float32)
+        self.dest_orog = set_up_variable_cube(
+            dest_orog, name='orography', units='m', spatial_grid='equalarea')
+
+        self.lapse_rate = set_up_variable_cube(
+            np.full((4, 4), DALR, dtype=np.float32), name='lapse_rate',
+            units='K m-1', spatial_grid='equalarea')
+
+        # specify temperature values ascending in 0.25 K increments
+        self.temperature = set_up_variable_cube(
+            np.linspace(276., 279.75, 16).reshape(4, 4).astype(np.float32),
```

Comment:
It would be easiest if this input data was written out as an array, just so it's easy to see the difference that the change to the orography is making when the specified lapse rate is applied.